packages/ddp-client/livedata_connection.js

```diff
@@ -1206,13 +1219,52 @@ _.extend(Connection.prototype, {
       // We'll now process and all of our buffered messages, reset all stores,
       // and apply them all at once.
       _.each(self._messagesBufferedUntilQuiescence, function (bufferedMsg) {
-        self._processOneDataMessage(bufferedMsg, updates);
+        self._processOneDataMessage(bufferedMsg, self._bufferedWrites);
       });
       self._messagesBufferedUntilQuiescence = [];
     } else {
-      self._processOneDataMessage(msg, updates);
+      self._processOneDataMessage(msg, self._bufferedWrites);
+    }
+
+    // Immediately flush writes when:
+    //  1. Buffering is disabled. Or;
+    //  2. any non-(added/changed/removed) message arrives.
+    var standardWrite = _.include(['added', 'changed', 'removed'], msg.msg);
+    if (self._bufferedWritesInterval === 0 || !standardWrite) {
+      self._flushBufferedWrites();
+      return;
+    }
+
+    if (self._bufferedWritesFlushAt === null) {
+      self._bufferedWritesFlushAt = new Date().valueOf() + self._bufferedWritesMaxAge;
+    }
+    else if (self._bufferedWritesFlushAt < new Date().valueOf()) {
+      self._flushBufferedWrites();
+      return;
+    }
+
+    if (self._bufferedWritesFlushHandle) {
+      clearTimeout(self._bufferedWritesFlushHandle);
+    }
+    self._bufferedWritesFlushHandle = setTimeout(self.__flushBufferedWrites,
+                                                      self._bufferedWritesInterval);
+  },
+
+  _flushBufferedWrites: function () {
+    var self = this;
+    if (self._bufferedWritesFlushHandle) {
+      clearTimeout(self._bufferedWritesFlushHandle);
+      self._bufferedWritesFlushHandle = null;
     }
 
+    self._bufferedWritesFlushAt = null;
+    self._performWrites(self._bufferedWrites);
```

Comment:
re more robust, I assume you mean before that commit?  Hmm, I mean, it doesn't call the after-update callbacks in the critical section, but it still invokes the "store API", which you could imagine could yield.

Correct that this whole concern is probably only a problem on the server, but people do use server-to-server DDP to some degree.